back/app/services/participation_context_service.rb

```diff
@@ -306,7 +306,16 @@ def mode_specific_idea_voting_disabled_reason(mode, context, user)
   end
 
   def posting_limit_reached?(context, user)
-    context.posting_limited? && context.ideas.where(author: user).size >= context.posting_limited_max
+    return true if context.posting_limited? && context.ideas.where(author: user).size >= context.posting_limited_max
+
+    # TODO: Check this works for 'everyone' surveys
+    if context.native_survey? && context.allow_anonymous_participation?
+      author_hash = Idea.create_author_hash user.id, context.id, true
+      # TODO: Needs an index on author_hash for this query
+      return context.ideas.where(author_hash: author_hash).size >= context.posting_limited_max
```

Comment:
Is it likely that the survey will be made anonymous half way through - in which case any survey submitted before it before the switch would not be counted here.